src/fides/api/ops/api/v1/endpoints/user_endpoints.py

```diff
@@ -191,3 +211,120 @@ def user_logout(
+    *,
+    db: Session = Depends(deps.get_db),
+    user_id: str,
+    systems: List[FidesKey],
+) -> List[SystemSchema]:
+    """
+    Endpoint to override the systems for which a user is "system manager".
+    All systems the user manages are replaced with those in the request body.
+    """
+    user = validate_user_id(db, user_id)
+
+    if len(set(systems)) != len(systems):
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=f"Cannot add user {user_id} as system manager. Duplicate systems in request body.",
+        )
+
+    retrieved_systems: Query = db.query(System).filter(System.fides_key.in_(systems))
+    if retrieved_systems.count() != len(systems):
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"Cannot add user {user_id} as system manager. System(s) not found.",
+        )
+
+    logger.info("Updating systems for which user {} is system manager", user_id)
+
+    # Adding new systems for which the user is not already a manager
+    for system in retrieved_systems:
+        if user not in system.users:
+            user.set_as_system_manager(db, system)
+
+    # Removing systems for which the user in no longer a manager
+    for system in user.systems:
+        if system not in retrieved_systems:
+            user.remove_as_system_manager(db, system)
+
+    return user.systems
+
+
+@router.get(
+    urls.SYSTEM_MANAGER,
+    dependencies=[Security(verify_oauth_client, scopes=[SYSTEM_MANAGER_READ])],
+    response_model=List[SystemSchema],
+)
+def get_user_systems(
+    *,
+    db: Session = Depends(deps.get_db),
+    user_id: str,
+) -> List[SystemSchema]:
+    """
+    Endpoint to retrieve all the systems for which a user is "system manager".
+    """
+    user = validate_user_id(db, user_id)
+    logger.info("Getting systems for which user {} is system manager", user_id)
+
+    return user.systems
+
+
+@router.get(
+    urls.SYSTEM_MANAGER_DETAIL,
+    dependencies=[Security(verify_oauth_client, scopes=[SYSTEM_MANAGER_READ])],
+    response_model=SystemSchema,
+)
+def get_user_system(
+    *, db: Session = Depends(deps.get_db), user_id: str, system_key: FidesKey
+) -> SystemSchema:
+    """
+    Endpoint to retrieve a single system managed by the given user.
+    """
+    system: System = get_system_by_fides_key(db, system_key)
+    user = validate_user_id(db, user_id)
+
+    if not system in user.systems:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"User {user_id} is not a manager of system {system.fides_key}",
+        )
+
+    logger.info(
+        "Getting system {} for which user {} is system manager",
+        system.fides_key,
+        user_id,
+    )
+
+    return system
+
+
+@router.delete(
+    urls.SYSTEM_MANAGER_DETAIL,
+    dependencies=[Security(verify_oauth_client, scopes=[SYSTEM_MANAGER_DELETE])],
+    status_code=HTTP_204_NO_CONTENT,
+)
+def remove_user_as_system_manager(
```

Comment:
am I understanding right that you could feasibly also just use `update_system_manager` instead of this endpoint? i.e. this endpoint would only be used if you didn't already know all the systems a user managed but wanted to remove one?

Reply:
Correct, you could use `update_system_manager` if you knew exactly which systems you wanted the user to have. I added this extra endpoint to just remove one in case it was useful, but I can remove if you didn't need it!

Reply:
heh I appreciate it, just wanted to check my understanding! I'll let you know—I'm not quite sure what I'll need yet 😅

Reply:
yeah no problem, happy to add/remove variations of these endpoints as needed.